### ISSUE TITLE
[Summary] Explanation Operations

### ISSUE THREAD

**Issue by nfelnlp** [DFKI-NLP/InterroLang#13]
<table>
<tr>
	<td> <b>Operation</b> </td>
        <td> <b>Terminals / Prompts</b> </td>
	<td> <b>Action</b> </td>
        <td> <b>Description</b> </td>
	<td> <b>Tools</b> </td>
        <td> <b>Status</b> </td>
</tr>
<tr>
        <td> <b>nlpattribute</b> </td>
        <td> <a href="https://github.com/nfelnlp/InterroLang/blob/main/prompts/explanation/local_feature_importance.txt">nlpattribute token | phrase | sentence {classes}</a> </td>
        <td> <a href="https://github.com/nfelnlp/InterroLang/blob/main/actions/explanation/feature_importance.py">feature_importance</a> </td>
        <td> Provides feature importances at the token (default), phrase or sentence level.  </td>
        <td> Captum (Integrated Gradients) </td>
        <td> ✅ </td>
</tr>
<tr>
        <td> <b>globaltopk</b> </td>
        <td> <a href="https://github.com/nfelnlp/InterroLang/blob/main/prompts/explanation/global_feature_importance.txt">important {number} {classes}</a> </td>
        <td> <a href="https://github.com/nfelnlp/InterroLang/blob/main/actions/explanation/global_topk.py">global_topk</a> </td>
        <td> Returns top k most attributed tokens across the entire dataset. </td>
        <td> Captum (Integrated Gradients) </td>
        <td> ✅ </td>
</tr>
<tr>
        <td> <b>nlpcfe</b> </td>
        <td> <a href="https://github.com/nfelnlp/InterroLang/blob/main/prompts/explanation/cfe.txt">nlpcfe {number}</a> </td>
        <td> <a href="https://github.com/nfelnlp/InterroLang/blob/main/actions/explanation/cfe.py">counterfactuals</a> </td>
        <td> Returns counterfactual explanations (model predicts another label) for a single instance. </td>
        <td> Polyjuice </td>
        <td> ✅ </td>
</tr>
<tr>
        <td> <b>adversarial</b> </td>
        <td> adversarial {number} </td>
        <td> </td>
        <td> Returns adversarial examples (model predicts wrong label) for a single instance. </td>
        <td> OpenAttack </td>
        <td> ✅ </td>
</tr>
<tr>
        <td> <b>similar</b> </td>
        <td> similar {number} </td>
        <td> <a href="https://github.com/nfelnlp/InterroLang/blob/main/actions/nlu/similarity.py">similarity</a> </td>
        <td> Gets number of training data instances that are most similar to the current one. </td>
        <td> Sentence Transformers</td>
        <td> ✅ </td>
</tr>
<tr>
        <td> <b>rules</b> </td>
        <td> rules {number} </td>
        <td> </td>
        <td> Outputs the decision rules for the dataset. </td>
        <td> Anchors </td>
        <td> ⛔ </td>
</tr>
<tr> 
        <td> <b>interact</b> </td>
        <td> interact </td>
        <td> </td>
        <td> Gets feature interactions. </td>
        <td> HEDGE </td>
        <td> ⛔ </td>
</tr>
<tr> 
        <td> <b>rationalize</b> </td>
        <td> rationalize </td>
        <td> <a href="https://github.com/nfelnlp/InterroLang/blob/main/actions/explanation/rationalize.py">rationalize</a> </td>
        <td> Explains the prediction for some specified instance in natural language.  </td>
        <td> Zero-shot prompting with GPTNeo parser </td>
        <td> ✅  </td>
</tr>
</table>
